Makefile

```diff
@@ -119,7 +123,31 @@ OCI_REGISTRY := projects.registry.vmware.com/tce
 
 ##### LINTING TARGETS
 .PHONY: lint mdlint shellcheck check yamllint misspell actionlint urllint imagelint
-check: ensure-deps lint mdlint shellcheck yamllint misspell actionlint urllint imagelint
+check:
+	@make_version=$(shell make --version | grep -E "GNU Make [0-9].[0-9]+" | grep -E -o "[0-9].[0-9]+" | cut -d "." -f1); \
+	if [ "$${make_version}" != "3" ]; then \
+		$(MAKE) check-parallel; \
+	else \
+		$(MAKE) check-serial; \
+	fi;
+
+.PHONY: check-serial
+check-serial:
+	@echo " "
+	@echo "***************************************************"
+	@echo "To speed up linting, update to the latest GNU Make."
+	@echo "***************************************************"
+	@echo " "
+	$(MAKE) lint mdlint shellcheck check yamllint misspell actionlint urllint imagelint
```

Comment:
Check is in the list of make targets to run here. This is going to recursion loop forever.

Reply:
Totally botched a rebase. Good catch.